src/nft/types.ts

```diff
@@ -0,0 +1,230 @@
+import type { ReactNode } from 'react';
+import type { Address, Hex, TransactionReceipt } from 'viem';
+import type { Call, TransactionError } from '../transaction/types';
+
+export type ContractType = 'ERC721' | 'ERC1155';
+
+/* Lifecycle Provider */
+
+export enum LifecycleType {
+  VIEW = 'view',
+  MINT = 'mint',
+}
+
+export type NFTLifecycleProviderReact = {
+  type: LifecycleType;
+  onError?: (error: NFTError) => void;
+  onStatus?: (lifecycleStatus: LifecycleStatus) => void;
+  onSuccess?: (transactionReceipt?: TransactionReceipt) => void;
+  children: ReactNode;
+};
+
+export type NFTLifecycleContextType = {
+  type: LifecycleType;
+  lifecycleStatus: LifecycleStatus;
+  updateLifecycleStatus: (status: LifecycleStatusUpdate) => void;
+};
+
+/* NFT Provider */
+
+export type NFTContextType = {
+  contractAddress: `0x${string}`;
+  tokenId?: string;
+} & NFTData;
+
+export type NFTProviderReact = {
+  children: ReactNode;
+  contractAddress: `0x${string}`;
+  tokenId?: string;
+  useNFTData: UseNFTData;
+};
+
+/* Mint Provider */
+
+export type NFTPrice = {
+  amount?: number;
+  currency?: string;
+  amountUSD?: number;
+};
+
+export type NFTMintContextType = {
+  quantity: number;
+  setQuantity: (quantity: string) => void;
+  buildMintTransaction: BuildMintTransaction;
+} & NFTMintData;
+
+export type NFTMintProviderReact = {
+  useNFTMintData: UseNFTMintData;
+  buildMintTransaction: BuildMintTransaction;
+  children: ReactNode;
+};
+
+type UseNFTData = (contractAddress: Hex, tokenId?: string) => NFTData;
+
+/**
+ * Note: exported as public Type
+ */
+export type NFTData = {
+  name?: string;
+  description?: string;
+  imageUrl?: string;
+  animationUrl?: string;
+  mimeType?: string;
```

Comment:
nit: Can we add a comment for this param?

Reply:
added.  it may change as I do more testing, but that's the current implementation.